src/tqec/compile/detectors/compute.py

```diff
@@ -494,13 +496,69 @@ def _compute_superimposed_template_instantiations(
     return ret
 
 
+def _extract_subtemplates_from_s3d(
+    s3d: npt.NDArray[numpy.int_],
+) -> list[npt.NDArray[numpy.int_]]:
+    """Extract 2D spatial subtemplates from a 3D array.
+
+    This function takes a 3D array where the first two dimensions
+    represent spatial coordinates and the third dimension represents time steps.
+    It extracts a list of 2D spatial arrays, one for each time step.
+
+    Args:
+        s3d: 3D numpy array representation of sub-templates.
+
+    Returns:
+        List of 2D numpy arrays, one for each time step.
+    """
+    return [s3d[:, :, i] for i in range(s3d.shape[2])]
+
+
+def _compute_detector_for_subtemplate(
+    args: tuple[
+        tuple[int, ...],
+        npt.NDArray[numpy.int_],
+        Sequence[Plaquettes],
+        Shift2D,
+        DetectorDatabase | None,
+        bool,
```

Comment:
I think the following would make sense here:
```suggestion
        tuple[int, ...],  # indices
        npt.NDArray[numpy.int_],  # s3d
        Sequence[Plaquettes],  # plaquettes
        Shift2D,  # increments
        DetectorDatabase | None,  # database
        bool,  # only_use_database
```